frontend/src/components/Forms/StatementForm.tsx

```diff
@@ -446,13 +450,13 @@ const StatementForm = (props: any) => {
           isPathBuilderComponent: true,
           InputIcon: DestinationIcon,
           onUpdate: async (selectedOption: string, formId: string) => {
-            const viaIndex = getConnectionId(formId, statement?.destinations);
+            const destinationIndex = getConnectionId(formId, statement?.destinations);
             const typeOption = selectedOption as TypeC11Enum;
-            if (viaIndex) {
-              api
-                .composerDestinationPartialUpdate(viaIndex, {
-                  type: typeOption,
-                })
+            if (destinationIndex) {
+              await api.composerDestinationPartialUpdate(destinationIndex, {
```

Comment:
I find it strange that this is not wrapped in a checkOwnership function. I suspect that the UI won't behave correctly when trying to change the type of a destination from a connectivity statement that I don't own. I didn't test this tho. cc @ddelpiano @Salam-Dalloul

Reply:
Good catch, I will open a card for @Salam-Dalloul for this. Thanks

Reply:
this has been fixed as well